Eplant/views/ChromosomeViewer/Viewer/Chromosome.tsx

```diff
@@ -0,0 +1,380 @@
+// -------
+// IMPORTS
+// -------
+import React, { FC, useEffect, useLayoutEffect, useState } from 'react'
+
+import { useCollections, useGeneticElements } from '@eplant/state'
+import { Unstable_Popup as Popup } from '@mui/base/Unstable_Popup'
+import ArrowLeft from '@mui/icons-material/ArrowLeft'
+import Box from '@mui/material/Box'
+import ClickAwayListener from '@mui/material/ClickAwayListener'
+import useTheme from '@mui/material/styles/useTheme'
+import Typography from '@mui/material/Typography'
+
+import {
+  CentromereItem,
+  ChromosomeItem,
+  GeneAnnotationItem,
+  GeneItem,
+} from '../types'
+
+import GeneAnnotation from './GeneAnnotation'
+import GeneList from './GeneList'
+
+//----------
+// TYPES
+//----------
+interface ChromosomeProps {
+  scale: number
+  chromosome: ChromosomeItem
+}
+
+interface Range {
+  start: number
+  end: number
+}
+
+//----------
+// COMPONENT
+//----------
+const Chromosome: FC<ChromosomeProps> = ({ scale, chromosome }) => {
+  // State
+  const [isHovered, setIsHovered] = useState<boolean>(false)
+  const [anchorOrigin, setAnchorOrigin] = useState<number[]>([])
+  const [anchorEl, setAnchorEl] = useState<null | any>(null)
+  const [geneRange, setGeneRange] = useState<Range>({
+    start: 0,
+    end: 0,
+  })
+  // Gene Annotation drawing
+  const [geneAnnotationArray, setGeneAnnotationArray] = useState<
+    GeneAnnotationItem[]
+  >([])
+
+  // Global State
+  const [geneticElements] = useGeneticElements()
+  const [collections] = useCollections()
+  const theme = useTheme()
+
+  // SVG drawing
+  const centromeres: CentromereItem[] = chromosome.centromeres
+  const hasCentromeres: boolean = centromeres.length > 0
+  const lastCentromereEnd: number = hasCentromeres
+    ? centromeres[centromeres.length - 1].end
+    : 0
+
+  const x: number = 10
+  const y: number = 0
+  const width: number = 10
+  const perBpHeight: number = 0.000015
+  let start: number = 0
+
+  // Gene List popover variables
+  const openPopup = Boolean(anchorEl)
+
+  // Fire before paint, converts geneticElements into geneAnnotationArray
+  useLayoutEffect(() => {
+    const poplar = false
+    const species = poplar ? 'Populus_trichocarpa' : 'Arabidopsis_thaliana'
+    let newGeneAnnotationArray: GeneAnnotationItem[] = []
+    geneticElements.map((gene) => {
```

Comment:
It seems that you are firing off an api call regardless of whether the geneticElement information has been stored already, and this is happening on every chromosome. I think this is logic that should occur only once at the top level component (the Viewer component) and the viewer can pass the relevant GeneItems to each chromosome. Ideally you would also only fire off an api call only for the new gene and not on every gene loaded (maintaining a set of gene ids should make this easy).

Let me know if this makes sense or if you'd like to hop on a quick call sometime to go over this.

Reply:
Yeah, that makes sense, it can definitely be moved to the top level.